### PR TITLE
[RunAllTests] Fix part of #2844: work around Bazel-specific CI issue that's causing an OpenJDK crash/flake

### PR DESCRIPTION
**Pull request by BenHenning**
Fix part of #2844.

This attempts to fix part of #2844 by normalizing test compiling/running on JDK 9 (for parity with Gradle tests--see #2845), and by working around what appears to be a resource constraint issue Bazel is hitting. For the latter, see https://github.com/bazelbuild/bazel/issues/3236#issuecomment-587752609. All configuration changes in this PR are done via changes to the CI environment's ``~/.bazelrc`` file, so nothing here should affect local builds.

Note that configuring Java versions in Bazel is complicated. In particular, at any time there can be 5 different versions of Java at play:
1. The version of Java used to build sources (defaults to a remote JDK for build reproducibility). Not sure what the default is here, but it seems to be 11.
2. The version of Java used to run tests/binaries (defaults to the local environment's JDK as determined by $JAVA_HOME).
3. The version of Java used to build tools used by Bazel. Defaults to a remote JDK 11.
4. The version of Java used to run tools used during the build. Defaults to local JDK 11 (if available--not sure what happens if it's not available; it might use the Java version in (5)).
5. The version of Java used to run Bazel itself. This defaults to an embedded JDK 11.

Now, from what I can tell, I think we only need to configure (1) since (2) should already be using the correct version of Java (JDK 9). We do want the majority of the build to be on JDK 9 since that's what we're running on, but this isn't *super* important. See also https://docs.bazel.build/versions/master/bazel-and-java.html#configuring-the-jdk & https://github.com/bazelbuild/bazel/issues/5594 for more details on how this Java resolution works.

Finally, similar to #2845 this cannot be observed as "correct" until we observe the flake itself disappearing over the next few weeks.

### CODE REVIEW

File: .github/workflows/build_tests.yml [hunk 1]
```diff
@@ -19,18 +19,39 @@ jobs:
...
+          $JAVA_HOME/bin/java -version
+
+      - name: Set up Bazel
+        uses: abhinavsingh/setup-bazel@v3
+        with:
+          version: 4.0.0
+
+      - name: Configure Bazel to use JDK 9 for building
+        run: |
+          echo "build --java_language_version=9" >> $HOME/.bazelrc
```
Comment (BenHenning) [inline .github/workflows/build_tests.yml:44]:
```suggestion
      - name: Configure Bazel to use JDK 9 for building
        run: |
          echo "build --java_language_version=9" >> $HOME/.bazelrc

      - name: Configure Bazel to use specific sandbox tmpfs
        run: |
          echo "build --enable_platform_specific_config" >> $HOME/.bazelrc
          echo "build:linux --sandbox_tmpfs_path=/tmp" >> $HOME/.bazelrc
```

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -62,21 +62,39 @@ jobs:
...
+
       - name: Set up Bazel
         uses: abhinavsingh/setup-bazel@v3
         with:
           version: 4.0.0
+          
+      - name: Configure Bazel to use JDK 9 for building
+        run: |
+          echo "build --java_language_version=9" >> $HOME/.bazelrc
+
```
Comment (BenHenning) [inline .github/workflows/unit_tests.yml]:
```suggestion

      - name: Configure Bazel to use specific sandbox tmpfs
        run: |
          echo "build --enable_platform_specific_config" >> $HOME/.bazelrc
          echo "build:linux --sandbox_tmpfs_path=/tmp" >> $HOME/.bazelrc

```

**Comment by vinitamurthi** [comment]
Is this still required?

**Comment by BenHenning** [comment]
@vinitamurthi this is expected to help, but it may not be sufficient to fully fix the Bazel-caused flakes.

**Review by vinitamurthi — APPROVED** [review]

**Review by anandwana001 — APPROVED** [review]
LGTM

**Comment by BenHenning** [comment]
Thanks all! Enabling auto-merge after resolving conflicts since some of the changes here may help with work I'm doing in #1861 to make Bazel CI runs a bit faster.

**Pull request merged**